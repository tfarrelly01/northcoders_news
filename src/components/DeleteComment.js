import React from 'react';
import PropTypes from 'prop-types';

const DeleteComment = props => (
  <div className="media-right">
    <a className="btn btn-danger has-text-danger" aria-label="Delete" onClick={props.handleDelete}>
      <i className="fa fa-trash-o fa-2x" aria-hidden="true" />
    </a>
  </div>
);

DeleteComment.defaultProps = {
  handleDelete: () => {}
};

DeleteComment.propTypes = {
	handleDelete: PropTypes.func.isRequired
};

export default DeleteComment;
