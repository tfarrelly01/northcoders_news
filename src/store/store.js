import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import reducer from '../reducer/reducer';

const logger = createLogger();

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk, logger),
    typeof window === 'object' && typeof window.devToolsExtension !== 'undefined'
      ? window.devToolsExtension()
      : f => f
  )
);

export default store;