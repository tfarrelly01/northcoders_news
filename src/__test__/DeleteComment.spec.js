import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../store/store';

import DeleteComment from '../components/DeleteComment';

describe('DeleteComment Component', () => {
  test('DeleteComment renders correctly', () => {
    const component = shallow(
      <Provider store={store}>
        <DeleteComment />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});