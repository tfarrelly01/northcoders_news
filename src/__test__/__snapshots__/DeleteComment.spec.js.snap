// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteComment Component DeleteComment renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <DeleteComment
    handleDelete={[Function]}
  />,
  "nodes": Array [
    <DeleteComment
      handleDelete={[Function]}
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <DeleteComment
          handleDelete={[Function]}
        />
      </Provider>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <DeleteComment
            handleDelete={[Function]}
          />,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <DeleteComment
          handleDelete={[Function]}
        />,
        "_debugID": 2,
        "_renderedOutput": <DeleteComment
          handleDelete={[Function]}
        />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <DeleteComment
      handleDelete={[Function]}
    />
  </Provider>,
}
`;
